xray/fasthttp.go

```diff
@@ -0,0 +1,94 @@
+package xray
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/aws/aws-xray-sdk-go/header"
+	"github.com/valyala/fasthttp"
+)
+
+// FastHTTPHandler
+type FastHTTPHandler struct {
+	cfg *Config
+}
+
+// NewFastHTTP returns a struct that provides Handle method
+// that satisfy fasthttp.RequestHandler interface.
+func NewFastHTTP(cfg *Config) *FastHTTPHandler {
+	return &FastHTTPHandler{
+		cfg: cfg,
+	}
+}
+
+// Handler wraps the provided fasthttp.RequestHandler
+func (h *FastHTTPHandler) Handler(sn SegmentNamer, handler fasthttp.RequestHandler) fasthttp.RequestHandler {
+	return func(ctx *fasthttp.RequestCtx) {
+		if h.cfg != nil {
+			ctx.SetUserValue(fasthttpContextConfigKey, h.cfg)
```

Comment:
It is not possible to use [xray/config.go#L133](https://github.com/aws/aws-xray-sdk-go/blob/master/xray/config.go#L133) because fasthttp does not work with contexts.

I needed to use this config [xray/fasthttp_test.go#L40](https://github.com/nicolascb/aws-xray-sdk-go/blob/feature/fashttp-handler/xray/fasthttp_test.go#L40) and one way I found doing this was by creating the `NewFastHTTP(cfg * Config)` method. 

I confess that it didn't look very elegant, do you have any suggestions on how we can improve this?
It may make sense to isolate the behavior of this method [xray/config.go#L133](https://github.com/aws/aws-xray-sdk-go/blob/master/xray/config.go#L133)  in an helper function and use it in `NewFastHTTP`.

Reply:
yes makes sense! couldn't find a good way to wrap config.